docs/api/api.md

```diff
@@ -1,212 +1,84 @@
-                    'callback'            => [ $this, 'get_items' ],
-                    'permission_callback' => [ $this, 'check_permission' ],
-                    'args'                => $this->get_collection_params(),
-                ]
-            ]
-        );
-    }
+use WeDevs\Dokan\Traits\VendorAuthorizable;
 
-    public function get_items( $request ) {
-        $items = []; // Your implementation
-        $total = count( $items );
-        
-        $response = rest_ensure_response( $items );
-        return $this->format_collection_response( $response, $request, $total );
-    }
+class VendorProductController extends \WC_REST_Products_Controller {
+    use VendorAuthorizable;
 }
 ```
 
-### Override Vendor REST Controller permission
-
-```php
-class ExampleVendorController extends DokanRESTVendorController {
-    // protected $namespace = 'dokan/v1'; (namespace will be inherited from the parent class)
-    protected $rest_base = 'example-vendor';
-
-    public function register_routes() {
-        register_rest_route(
-            $this->namespace,
-            '/' . $this->rest_base,
-            [
-                [
-                    'methods'             => WP_REST_Server::READABLE,
-                    'callback'            => [ $this, 'get_items' ],
-                    'permission_callback' => [ $this, 'check_permission' ],
-                    'args'                => $this->get_collection_params(),
-                ]
-            ]
-        );
-    }
-    
-    public function check_permission() {
-        // Custom permission check.
-        return current_user_can( 'dokandar' ) || is_user_logged_in();
-    }
+This trait provides a `check_permission` method, which can be used to verify if a user has the vendor(`dokandar`) capabilities to perform a given action.
 
-    public function get_items( $request ) {
-        $items = []; // Your implementation
-        $total = count( $items );
-        
-        $response = rest_ensure_response( $items );
-        return $this->format_collection_response( $response, $request, $total );
-    }
-}
-```
+## Example Implementations
 
-### Customer REST Controller Implementation
+### Admin Controller Example
 
 ```php
-abstract class DokanRESTCustomerController extends DokanRESTBaseController {
-    // protected $namespace = 'dokan/v1'; (namespace will be inherited from the parent class)
-    protected $rest_base = 'customer';
-    
-    public function check_permission() {
-        return is_user_logged_in();
-    }
-}
-```
-
-### How to extend the Customer REST Controller
+use WeDevs\Dokan\REST\DokanBaseAdminController;
 
-```php
-class ExampleCustomerController extends DokanRESTCustomerController {
-    // protected $namespace = 'dokan/v1'; (namespace will be inherited from the parent class)
-    protected $rest_base = 'example-customer';
+class ExampleAdminController extends DokanBaseAdminController {
+    protected $namespace = 'dokan/v1';
```

Comment:
Can we add the default namespace in doc which is already available in the `DokanBaseAdminController` class...? (Optional)